olive/passes/olive_pass.py

```diff
@@ -305,94 +378,13 @@ def _resolve_config(
     def _initialize(self):
         """Initialize the pass. Pass specific initialization should be done here."""
 
-    def search_space(self) -> Dict[str, SearchParameter]:
```

Comment:
Thanks for fixing this! When I first did the new design implementation, for some reason, I was really into using private attributes haha 
Regretted doing that so these code cleanup is great!